### PR TITLE
docs(browser-support): document browser support and polyfills

### PR DESCRIPTION
**Pull request by mlaval**
This is a first try at documenting browser support an polyfills.
The content needs to be reviewed and validated.

/cc @wardbell @IgorMinar @naomiblack

### CODE REVIEW

File: public/docs/ts/latest/guide/browser-support.jade [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    td
+    td
+    td
+    td
+    td Jelly Bean<br>(4.1, 4.2, 4.3)
+    td
+    
+    
+:marked
+  To ensure this compatibility, the unit tests of the framework are run on all of them for every single commit.
```
Comment (IgorMinar) [inline public/docs/ts/latest/guide/browser-support.jade]: "every single commit" sounds a bit dramatic and it's not really true. we run tests on PRs which can contain multiple commits.

File: public/docs/ts/latest/guide/browser-support.jade [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+:marked
+  To ensure this compatibility, the unit tests of the framework are run on all of them for every single commit. 
+  This is done as part of the continuous integration setup of the project, using [SauceLabs](https://saucelabs.com/) and [Browserstack](https://www.browserstack.com).
+  
+  # Polyfills #
+  Angular is built on the latest standards of the web platform.
+  Targeting such a wide range of browsers can be challenging as they may not support all the functionalities of modern browsers.
+  
+  
+  To fill the gaps, the best solution is to use polyfills which "upgrade" old browsers by adding missing functionalities.
```
Comment (IgorMinar) [inline public/docs/ts/latest/guide/browser-support.jade]: functionalities -> features

File: public/docs/ts/latest/guide/browser-support.jade [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+  For each browser, some polyfill may be required to run any application, and some may be needed only when some features are used.
+
+.alert.is-helpful 
+  span As a developer, depending on the browsers you target and on the features you use, you'll need to load a set of polyfills.
+  span  The tables below will help you making the right choices.
+  
+.alert.is-important 
+  span The polyfills described here under are the ones which, others may be required for your application.
```
Comment (IgorMinar) [inline public/docs/ts/latest/guide/browser-support.jade]: "here under" -> below

Comment (IgorMinar) [inline public/docs/ts/latest/guide/browser-support.jade]: "are the ones which, others may be required for your application" - what do you mean by this? I don't understand. can you rephrase?

File: public/docs/ts/latest/guide/browser-support.jade [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+    td Safari 7 & 8, IE10 & 11, Android 4.1+
+    td ES6
+  tr
+    td IE9
+    td ES6<br>classList
+    
+    
+:marked
+  ## Optional polyfills ##
+  On top of the mandatory polyfills, some features of Angular may require some extra polyfills.
```
Comment (IgorMinar) [inline public/docs/ts/latest/guide/browser-support.jade]: "on top" -> "In addition"

File: public/docs/ts/latest/guide/browser-support.jade [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+      a(href="../api/common/index/DecimalPipe-class.html") decimal
+      span  and 
+      a(href="../api/common/index/PercentPipe-class.html") percent
+      span  pipes
+    td Intl API
+    td All but Chrome, Firefox, Edge and IE11
+  tr
+    td 
+      a(href="../api/common/index/NgClass-directive.html") NgClass
+      span  on SVG elements
```
Comment (IgorMinar) [inline public/docs/ts/latest/guide/browser-support.jade]: is the double space after span intentional?

File: public/docs/ts/latest/guide/browser-support.jade [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+    td MIT / Unicode licence
+    td 13.5KB
+  tr
+    td Web Animations
+    td
+      a(href="https://github.com/web-animations/web-animations-js") https://github.com/web-animations/web-animations-js
+    td Apache
+    td 14.8KB
+  tr
+    td Typed Arrays
```
Comment (IgorMinar) [inline public/docs/ts/latest/guide/browser-support.jade]: really? what requires typed arrays?

File: public/docs/ts/latest/guide/browser-support.jade [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+  tr
+    td Safari 7 & 8, IE10 & 11, Android 4.1+
+    td ES6
+  tr
+    td IE9
+    td ES6<br>classList
+    
+    
+:marked
+  ## Optional polyfills ##
```
Comment (IgorMinar) [inline public/docs/ts/latest/guide/browser-support.jade]: "Optional browser features to polyfill" - isn't this more accurate? the list above doesn't list polyfills but feature that should be polyfilled.

File: public/docs/ts/latest/guide/browser-support.jade (hunk 6 above)
Comment (IgorMinar) [inline public/docs/ts/latest/guide/browser-support.jade]: the same for Blob and FormData - they are not really needed unless you are using particular browser features in your app at which point you are responsible for figuring out what polyfills you need. none of these should be used directly by angular's code unless I'm missing something.

File: public/docs/ts/latest/guide/browser-support.jade (hunk 5 above)
Comment (mlaval) [inline public/docs/ts/latest/guide/browser-support.jade]: Yes, for formating purpose

File: public/docs/ts/latest/guide/browser-support.jade (hunk 6 above)
Comment (mlaval) [inline public/docs/ts/latest/guide/browser-support.jade]: We use typed arrays in http module, and we have tests in the same module that use Blob and FormData. That's why I documented this feature here.
I could be removed if too specific.

**Comment by mlaval** [comment]
Thanks for the suggestions, I updated the PR

**Comment by IgorMinar** [comment]
LGTM!

@wardbell can we merge this in?

**Comment by wardbell** [comment]
Ill have a chance to review when I return on Monday. @foxandxss maybe you can get a jump on this and make sure it is ready to go. Thanks everyone!

**Comment by Foxandxss** [comment]
Yes sir, will do today.

**Comment by Foxandxss** [comment]
I would change the "ES6" to say "ES6 shim" (I know that perhaps your way is more correct, but thinking about the user perspective, I think they are more familiar with the "ES6 shim" term.

On the other hand, the first time is a bit overcomplicated. Instead of listing all versions one by one, can't we say IE9+, Android4.1+, etc?

**Comment by mlaval** [comment]
"ES6 shim" is also the name of a well-known library, it could be confusing, especially since we suggest to use core-js at the end (to shim ES6 :D).
Anyway, it should be ES2015. So it could be changed to "ES2015 shim".

About the first table, the intention was to follow the format of a SauceLabs badge. To me, it looks better than saying: 
Angular supports most recent browsers. This includes the Chrome (latest), Firefox (latest), Edge 13+, IE 9+, Safari 7+, iOS 7+, Android 4.1+ and IE mobile 11.
It can definitely be changed.

**Comment by Foxandxss** [comment]
> "ES6 shim" is also the name of a well-known library, it could be confusing, especially since we suggest to use core-js at the end (to shim ES6 :D).
Anyway, it should be ES2015. So it could be changed to "ES2015 shim".

Agreed.

> It can definitely be changed.

It is only my opinion, let see what the others say

**Comment by IgorMinar** [comment]
I'm merging this because we need other pages to link to it, we can improve it further next week if needed
